### ISSUE TITLE
[java] LambdaCanBeMethodReference: False positive with array creation containing constructor call in receiver

### ISSUE THREAD

**Issue by mmnhgo** [pmd/pmd#6709]
**Affects PMD Version:** 7.24.0

**Rule:**

[LambdaCanBeMethodReference](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#lambdacanbemethodreference)

**Description:**
The rule reports a **false positive** for a lambda whose method-call receiver contains an array creation expression with a nested constructor call.
The official documentation explicitly states that the rule will not report lambdas of the form `x -> new CtorCall().something(x)` because converting to a method reference would change the object creation timing (one object at method-ref creation time vs. one object per lambda invocation).
This reproducer has exactly the same semantic issue, but the constructor call is hidden inside an array initializer (`new Holder[] { new Holder() }[0]`). The current implementation in `canBeTransformed()` only excludes a top-level `ASTConstructorCall` and therefore incorrectly suggests the transformation.

**Code Sample demonstrating the issue:**

```java
import java.util.function.IntPredicate;

public class Repro {
    static IntPredicate p() {
        return x -> new Holder[] { new Holder() }[0].test(x); //<-should not report
    }

    static class Holder {
        boolean test(int x) {
            return x > 0;
        }
    }
}
```

**Expected outcome:**
PMD should not report any violation.
The suggested method reference `new Holder[]{ new Holder() }[0]::test `changes the program semantics (array + `Holder` are created only once instead of on every lambda invocation). This is a false positive.

**Running PMD through:** *CLI*
